### PR TITLE
add case for blockcommit with relative path

### PR DESCRIPTION
**Pull request by nanli1**
VIRT-293737: Do blockcommit with --keep-relative and relative path continuously
Signed-off-by: nanli <nanli@redhat.com>
**Depend on** : https://github.com/avocado-framework/avocado-vt/pull/3484
**Test result:** 
```
 /usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 backingchain.blockcommit.relative_path
JOB ID     : e142deb055b7cd664f01e95b43a40bce4f7f71d6
JOB LOG    : /root/avocado/job-results/job-2022-08-05T16.04-e142deb/job.log
 (1/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.file_disk.keep_relative.active.vm_started: PASS (20.65 s)
 (2/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.file_disk.keep_relative.active.vm_paused: PASS (22.04 s)
 (3/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.file_disk.keep_relative.inactive.vm_started: PASS (20.61 s)
 (4/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.file_disk.keep_relative.inactive.vm_paused: PASS (20.71 s)
 (5/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.block_disk.keep_relative.active.vm_started: PASS (50.29 s)
 (6/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.block_disk.keep_relative.active.vm_paused: PASS (50.17 s)
 (7/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.rbd_with_auth_disk.keep_relative.active.vm_started: PASS (24.15 s)
 (8/8) type_specific.io-github-autotest-libvirt.backingchain.blockcommit.relative_path.rbd_with_auth_disk.keep_relative.active.vm_paused: PASS (25.35 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

### CODE REVIEW

**Comment by nanli1** [comment]
@chloerh hi haijiao,Could you please help review this pr

**Review by chloerh — APPROVED** [review]

File: provider/virtual_disk/disk_base.py [hunk 1]
```diff
@@ -221,3 +264,96 @@ def set_disk_xml(disk_dict):
...
+        """
+        Prepare relative path
+
+        :params disk_type: disk type
+        :return: dest image for new disk and backing list
+        """
+        dst_img, backing_list = '', []
+        if disk_type == "file":
+            libvirt.create_local_disk(
+                "file", path=self.base_dir+"/a/a", size="500M",
```
Comment (dzhengfy) [inline provider/virtual_disk/disk_base.py:278]: `path=self.base_dir + "/a/a"`

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: provider/virtual_disk/disk_base.py (hunk 1 above)
Comment (nanli1) [reply provider/virtual_disk/disk_base.py]: Thanks dzheng, done . :D

**Comment by nanli1** [comment]
modular_redesign

**Pull request merged**